client/cypress/e2e/add-user.cy.ts

```diff
@@ -1,17 +1,9 @@
 import { User } from 'src/app/users/user';
 import { AddUserPage } from '../support/add-user.po';
 
-describe('Add user', () => {
-  const page = new AddUserPage();
-
-  beforeEach(() => {
-    page.navigateTo();
-  });
-
-  it('Should have the correct title', () => {
-    page.getTitle().should('have.text', 'New User');
-  });
+const page = new AddUserPage();
 
+const checkAddUserButtonState = () => {
```

Comment:
@NicMcPhee I'm with @floogulinc, I'm afraid—this refactoring looks odd to me. 🤷🏻‍♂️

I would never expect a named function to declare a test case inside of it. That makes me have to jump around the source file to see what's actually being tested.

The problem is, BCH doesn't understand that Cypress, Jasmine, etc use a DSL. 🫤 A `describe(() => {})` block in Cypress is not the same conceptual object as a function, and there is no universe in which they should be held to the same length limits.

I don't know—does that make sense? 🤷🏻‍♂️

Reply:
It makes total sense, and I shared your concerns as I was doing the refactoring. I _like_ the `describe()/it()` structure that we get with Jasmine, and extracting those functions really didn't seem to improve readability for all the reasons you gave.

I'll unwind all that refactoring, and then we can see how we feel about things.

Reply:
Roger that, thanks! 🙂